enhancements/network-observability.md

```diff
@@ -0,0 +1,367 @@
+The web console will provide a NetFlow table showing traffic between
+pods.  In the future, more visualization and functionality will be
+provided including areas such as topology, network-related eBPF data,
+policy validation, security risks, and more.
+
+***Note:*** *The term "NetFlow" is used generically throughout this document and
+is synonymous with IPFIX, which is the IETF version of NetFlow.*
+
+### User Stories
+
+1. As a cluster/network admin, I need access to data that allows me to debug network problems and conduct audits of network traffic.
+    - “I’m trying to target traffic towards a specific pod, and the traffic coming into the cluster isn’t getting there.”
+    - “The traffic coming into the cluster isn’t landing where I expect, where do things go wrong?”
+    - “I’m trying to understand why traffic isn’t flowing from pod to pod in the way that I expected.”
+
+2. As a cluster/network admin, I need basic search and filtering tools to analyze a subset of NetFlow data inside the OpenShift console.
+    - Users can filter NetFlow based on traffic source or destination (ie, view all of the traffic associated with a particular node, service, namespace, pod, etc.).
+
+
+### Implementation Details/Notes/Constraints
+
+Here are some of the limitations and constraints.
+
+1. CNI must be OVN-Kubernetes<br>
+    The network type (CNI) has to be OVN-Kubernetes since configuring OVS to
+export IPFIX data is only supported there.  By default, the CNI is OpenShift
+SDN and the overwhelming majority of customers run OpenShift SDN today.
+Therefore in a brown field scenario, this would require the user to change
+their CNI type before network observability can be enabled.
+    Note that the effort to implement OpenShift SDN to configure OVS is
+doable.  However, new features recently added have only been implemented
+in OVN-Kubernetes.
+
+2. Data sampling<br>
+    In order to scale in terms of collecting data, it is not uncommon to take
+a very small sample of data. By default, the data sample rate in OVS is
+400 to 1.  Networks with larger bandwidth, sample at an even higher rate
+such as 1,024 to 1.
+
+    With sampling, you will be able to get trends about your network.  If
+you are looking for a specific flow of data, it's likely not to be there.
+The total number of bytes and packets are just estimations when sampling
+is used.  Therefore if you need full visibility of the traffic and accuracy,
+the only choice is not do sampling.
+
+3. Resources required<br>
+    Exporting data is potentially CPU, memory and bandwidth-intensive, and
+typically requires a large amount of storage.  The CPU/memory/bandwidth
+resources can be managed by the sampling data rate.  The storage requirements
+can be controlled by the data retention policy or how long to keep the data.
+
+    Nevertheless, in order to maintain the same performance that you have
+prior to enabling network observability, will require additional resources.
+
+
+### Risks and Mitigations
```

Comment:
It seems risks have been listed, but not mitigations for those risks?

Reply:
Add mitigations in fab47eb